ocaml/xapi-storage-script/main.ml

```diff
@@ -650,8 +648,11 @@ module Attached_SRs = struct
         in
         let dir = Filename.dirname path in
         Sys.mkdir_p dir >>= fun () -> Sys.save path ~contents
-    )
-    >>= fun () -> return ()
+
+  let add smapiv2 plugin uids =
+    let key = Storage_interface.Sr.string_of smapiv2 in
+    Base.Hashtbl.set !sr_table ~key ~data:{sr= plugin; uids} ;
+    update_state_path () >>= fun () -> return ()
```

Comment:
Isn't that last part redundant? What is the purpose of `>>= fun () -> return ()`, can it be dropped?

Reply:
It's not redundant, add need to return an Lwt_result.t, but  update_state_path only returns an Lwt.t